source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -180,81 +170,71 @@ Note:
 will be similar to the upload-ova process:
 
 ![](../../../../images/wiki/import-ova.png)
 
-1. The front end sends the back end a VM and a path on a host.
-2. The back end retrieves the OVF configuration from the OVA using an
-   agent located on the host (Imageio/Other).
-3. The back end parses the OVF configuration and returns a VM to the
-   front end.
-4. The front end sends (optionally customized) parameters of the import
-   process to the back end.
-5. The back end persists the VM into the database.
-6. The back end triggers the creation of the target VM disks on the
+1. The frontend sends the backend a VM and a path on a host.
+2. The backend retrieves the OVF configuration from the OVA using vdsm.
+   vdsm will run the ova-ovirt tool.
+3. The backend parses the OVF configuration and returns a VM to the
+   frontend.
+4. The frontend sends (optionally customized) parameters of the import
+   process to the backend.
+5. The backend persists the VM into the database.
+6. The backend triggers the creation of the target VM disks on the
    storage domain.
-7. The back end sends the import parameters, including the destination
-   disks, to imageio-proxy, that sends them to imageio.
-8. Imageio reads the disk from the OVA and according to the import
-   parameters streams the data into the corresponding target disks.
+7. The backend sends the import parameters, including the destination
+   disks to vdsm.
+8. vdsm runs the ova-ovirt tool the reads the disk from the OVA and
+   according to the import parameters streams the data into the 
+   corresponding target disks.
 
 ## Downloading an OVA
 This part depends on the implementation of the download-image flow. In
 theory, it should look like this:
 
 ![](../../../../images/wiki/download-ova.png)
 
-1. The front end passes the back end a VM (or template) to download.
-2. The back end sends import parameters to imageio-proxy in order to
-   prepare to download-ova.
-3. The URL that is returned by imageio-proxy is returned to the front
-   end through the back end.
-4. The front end initializes a target file to write the OVA to, and
-   starts listening on the URL.
-5. The back end passes the disk to imageio-proxy to copy, and
-   imageio-proxy delivers it to imageio.
-6. Imageio starts reading the disk - converting it to QCOW - and sends
-   it to the URL. The front end then streams the data into the target
-   file.
-7. The back end sends the OVF configuation to imageio-proxy.
-   Imageio-proxy sends it to imageio, and imageio writes the OVF
-   configuration to the URL. The front end then streams the data into
-   the target file.
+1. The frontend passes the backend a VM (or template) to download.
+2. Exporting to an OVA on the host or shared storage (as describe in
+   'Exporting a VM or Template as an OVA')
+3. Frontend displays a link for downloading the OVA.
+4. Press on the download link will start an ovirt-imageio download
+   to the browser (not implemented yet in the frontend).
```

Comment:
If we don't have a way to stream data from the disks, yes, we need to create the ova somewhere, and then we can download it. If we find a way to stream the data, we will have to rethink the flow.

Daniel, I think we should add a note that this solution allows resuming a download at any point, while a streaming solution will most likely require starting the entire export/convert/download from scratch.